src/guide/2-model-definition/3-agent.rst

```diff
@@ -1,51 +1,139 @@
 Defining Agents
 ===============
 
-Agents are the central component of FLAME GPU simulations, as they map effectively to the highly parallel architecture of GPUs.
+Agents are the central component of FLAME GPU simulations, and are directly equivalent to agent-based modelling agents. However, 
+they can also be used to represent other things such as scalar fields across the environment. Agents are represented by an ``AgentDescription``
+object
 
-**TODO General Description**
+Defining a New Agent Type
+-------------------------
+FLAMEGPU2 agents are associated with a particular model. As such they are created via a `ModelDescription` object and are initialised with a name:
 
-Model Definition
+.. tabs::
+  
+  .. code-tab:: python
+
+    # Create a new agent called 'predator' associated the model 'model' 
+    predator = model.newAgent("predator")
+ 
+  .. code-tab:: cpp
+
+    // Create a new agent called 'predator' associated the model 'model' 
+    AgentDescription &predator = model.newAgent("predator");
+
+Agent Properties
```

Comment:
Agents/messages have variables, environment has properties.

We use the distinction, because env properties can never be modified on device.